integrations/pjrt/src/iree_pjrt/rocm/client.cc

```diff
@@ -17,21 +19,68 @@ ROCMClientInstance::ROCMClientInstance(std::unique_ptr<Platform> platform)
   // TODO: Get this when constructing the client so it is guaranteed to
   // match.
   cached_platform_name_ = "iree_rocm";
-  IREE_CHECK_OK(iree_hal_rocm_driver_module_register(driver_registry_));
 }
 
 ROCMClientInstance::~ROCMClientInstance() {}
 
 iree_status_t ROCMClientInstance::CreateDriver(iree_hal_driver_t** out_driver) {
-  iree_string_view_t driver_name = iree_make_cstring_view("rocm");
-  IREE_RETURN_IF_ERROR(iree_hal_driver_registry_try_create(
-      driver_registry_, driver_name, host_allocator_, out_driver));
+  iree_string_view_t driver_name = iree_make_cstring_view("hip");
+
+  // Device params.
+  iree_hal_hip_device_params_t default_params;
+  iree_hal_hip_device_params_initialize(&default_params);
+
+  // Driver params.
+  iree_hal_hip_driver_options_t driver_options;
+  iree_hal_hip_driver_options_initialize(&driver_options);
+
+  IREE_RETURN_IF_ERROR(iree_hal_hip_driver_create(driver_name, &driver_options,
+                                                  &default_params,
+                                                  host_allocator_, out_driver));
   logger().debug("ROCM driver created");
+
+  // retrieve the target name of current available device
+  iree_host_size_t device_info_count;
+  iree_hal_device_info_t* device_infos;
+  IREE_RETURN_IF_ERROR(iree_hal_driver_query_available_devices(
+      *out_driver, host_allocator_, &device_info_count, &device_infos));
+
+  // TODO: here we just use the target name of the first available device,
+  // but ideally we should find the device which will run the program
+  if (device_info_count > 0) {
+    hipDeviceProp_tR0000 props;
+    IREE_RETURN_IF_ERROR(iree_hal_hip_get_device_properties(
+        *out_driver, device_infos->device_id, &props));
+
+    // `gcnArchName` comes back like gfx90a:sramecc+:xnack- for a fully
+    // specified target. However the IREE target-chip flag only expects the
+    // prefix. refer to
+    // https://github.com/iree-org/iree-turbine/blob/965247e/iree/turbine/runtime/device.py#L495
+    std::string_view target = props.gcnArchName;
+    if (auto pos = target.find(':'); pos != target.npos) {
+      target = target.substr(0, pos);
+    }
+
+    hip_target_ = target;
+    logger().debug("HIP target detected: " + hip_target_);
```

Comment:
> ```
> + python test/test_add.py
> WARNING:jax._src.xla_bridge:Platform 'iree_rocm' is experimental and not all JAX functionality may be correctly supported!
> [IREE-PJRT] DEBUG: Using IREE compiler binary: /home/esaimana/actions-runner/_work/iree/iree/.venv/lib/python3.11/site-packages/iree/compiler/_mlir_libs/libIREECompiler.so
> [IREE-PJRT] DEBUG: Compiler Version: 3.1.0.dev+da45650503b955e[35](https://github.com/iree-org/iree/actions/runs/12031928938/job/33543044454?pr=19279#step:7:36)a45ea8e34b5f10b30a2f912 @ da45650503b955e35a45ea8e34b5f10b30a2f912 (API version 1.4)
> [IREE-PJRT] DEBUG: Partitioner was not enabled. The partitioner can be enabled by setting the 'PARTITIONER_LIB_PATH' config var ('IREE_PJRT_PARTITIONER_LIB_PATH' env var)
> [IREE-PJRT] DEBUG: ROCM driver created
> [IREE-PJRT] DEBUG: HIP target detected: gfx1100
> iree/runtime/src/iree/hal/drivers/hip/native_executable.c:306: UNKNOWN; HIP driver error 'hipErrorFileNotFound' (301): file not found; mismatched target chip? missing/wrong bitcode directory?; while invoking native function hal.executable.create; while calling import; 
> [ 0] bytecode jit__add.__init:1284 "jit(_add)/jit(main)/add"("<module>"(/home/esaimana/actions-runner/_work/iree/iree/integrations/pjrt/test/test_add.py:9:6))
> build_tools/testing/run_jax_tests.sh: line 31: 888283 Aborted                 (core dumped) JAX_PLATFORMS=$actual_jax_platform python $test_py_file > $actual_tmp_out
> ```
> 
> Hmm the error in rocm plugin comes with a hint (`mismatched target chip? missing/wrong bitcode directory?`) but it seems weird that I've already add some new logic to detect the HIP target and pass to IREE (via `--iree-hip-target`) before the compilation phase.
> 
> I'll investigate these errors in these days.

For test workflows, we typically set an environment variable explicitly, matched with the hardware installed into the runner chosen using the `runs-on` property:

https://github.com/iree-org/iree/blob/2e2c109e8d58c8964ad7a3ba26d6addbfefbc1c4/.github/workflows/pkgci_test_amd_w7900.yml#L31

In PJRT, matching the code from iree-turbine that detects available devices makes sense to me, instead of that environment variable approach. This is a JIT scenario, where the compiler is being used to generate code for the _current device_, not an arbitrary deployment target.

Can you have the PJRT client log the compile command used, including all flags?

Reply:
Sure! I'll dump more information for debugging.